src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKey.StandardProperties.cs

```diff
@@ -215,6 +224,7 @@ public int KeySize
                     }
                 }
 
+                _cachedKeySize = keySize;
```

Comment:
Lines 218-223 have early returns. Do we want to change the switch to assign `keySize`? Otherwise the Platform Crypto Provider won't benefit from the cache.

Reply:
Good catch.  I remember looking exactly at that and thinking it needed something special, but somehow the flow there caught me as just assigning the local... which, of course, isn't what it does.